pkg/hive/job/group.go

```diff
@@ -0,0 +1,473 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package job
```

Comment:
Yea, I didn't do that at first since `jobgroup.Group` stutters, so `job.Group` and `job.Collection` seemed more natural. But I don't mind `jobgroup` that much.